strncmp fatal error, missing 3rd argument with php8.2

### Expected behaviour

### Actual behaviour

Compilation says that strncmp needs 3 arguments instead of 2.
In redis.c there are lines that include this  :   strncmp(resp, ZEND_STRL("+OK"))
It should be replaced by strncmp(resp, "+OK", 3)

As far as I noticed there are 3 occurences in redis.c

### I'm seeing this behaviour on
- OS: CentOs
- Redis: 6.1.0
- PHP: 8.2
- phpredis:

### Steps to reproduce, backtrace or example script
Compiling on Centos with CPanel

### I've checked
- [ ] There is no similar issue from other users
- [ ] Issue isn't fixed in `develop` branch

I tried on another server and get the error, here is the output :

WARNING: channel "pecl.php.net" has updated its protocols, use "pecl channel-update pecl.php.net" to
 update
downloading redis-6.1.0.tgz ...
Starting to download redis-6.1.0.tgz (373,740 bytes)
.............................................................................done: 373,740 bytes
43 source files, building
running: phpize
Configuring for:
PHP Api Version:         20220829
Zend Module Api No:      20220829
Zend Extension Api No:   420220829
enable igbinary serializer support? [no] : enable lzf compression support? [no] : enable zstd compre
ssion support? [no] : enable msgpack serializer support? [no] : enable lz4 compression? [no] : use s
ystem liblz4? [yes] : building in /root/tmp/pear/pear-build-rootaQcBse/redis-6.1.0
running: /root/tmp/pear/redis/configure --with-php-config=/opt/cpanel/ea-php82/root/usr/bin/php-conf
ig --enable-redis-igbinary=no --enable-redis-lzf=no --enable-redis-zstd=no --enable-redis-msgpack=no
 --enable-redis-lz4=no --with-liblz4
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for a sed that does not truncate output... /bin/sed
checking for pkg-config... /bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking for system library directory... lib
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking for PHP prefix... /opt/cpanel/ea-php82/root/usr
checking for PHP includes... -I/opt/cpanel/ea-php82/root/usr/include/php -I/opt/cpanel/ea-php82/root
/usr/include/php/main -I/opt/cpanel/ea-php82/root/usr/include/php/TSRM -I/opt/cpanel/ea-php82/root/u
sr/include/php/Zend -I/opt/cpanel/ea-php82/root/usr/include/php/ext -I/opt/cpanel/ea-php82/root/usr/
include/php/ext/date/lib
checking for PHP extension directory... /opt/cpanel/ea-php82/root/usr/lib64/php/modules
checking for PHP installed headers prefix... /opt/cpanel/ea-php82/root/usr/include/php
checking if debug is enabled... no
checking if zts is enabled... no
checking for gawk... gawk
checking whether to enable redis support... yes, shared
checking whether to enable sessions... yes
checking whether to enable json serializer support... yes
checking whether to enable igbinary serializer support... no
checking whether to enable msgpack serializer support... no
checking whether to enable lzf compression... no
checking use system liblzf... no
checking whether to enable Zstd compression... no
checking use system libzstd... yes
checking whether to enable lz4 compression... no
checking use system liblz4... yes
checking for hash includes... /opt/cpanel/ea-php82/root/usr/include/php
checking for json includes... /opt/cpanel/ea-php82/root/usr/include/php
checking for redis json support... enabled
checking for redis igbinary support... disabled
checking for pkg-config... /bin/pkg-config
checking for git... yes
checking how to print strings... printf
checking for a sed that does not truncate output... (cached) /bin/sed
checking for fgrep... /bin/grep -F
checking for ld used by cc... /bin/ld
checking if the linker (/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /bin/nm -B
checking the name lister (/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 3458764513820540925
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert
_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop

checking for /bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking for gawk... (cached) gawk
checking command to parse /bin/nm -B output from cc object... ok
checking for sysroot... no
checking for mt... no
checking if : is a manifest tool... no
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if cc supports -fno-rtti -fno-exceptions... no
checking for cc option to produce PIC... -fPIC -DPIC
checking if cc PIC flag -fPIC -DPIC works... yes
checking if cc static flag -static works... yes
checking if cc supports -c -o file.o... yes
checking if cc supports -c -o file.o... (cached) yes
checking whether the cc linker (/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
configure: patching config.h.in
configure: creating ./config.status
config.status: creating config.h
config.status: executing libtool commands
running: make
/bin/sh /root/tmp/pear/pear-build-rootaQcBse/redis-6.1.0/libtool --tag=CC --mode=compile cc -I. -I/r
oot/tmp/pear/redis -I/root/tmp/pear/pear-build-rootaQcBse/redis-6.1.0/include -I/root/tmp/pear/pear-
build-rootaQcBse/redis-6.1.0/main -I/root/tmp/pear/redis -I/opt/cpanel/ea-php82/root/usr/include/php
 -I/opt/cpanel/ea-php82/root/usr/include/php/main -I/opt/cpanel/ea-php82/root/usr/include/php/TSRM -
I/opt/cpanel/ea-php82/root/usr/include/php/Zend -I/opt/cpanel/ea-php82/root/usr/include/php/ext -I/o
pt/cpanel/ea-php82/root/usr/include/php/ext/date/lib -I/opt/cpanel/ea-php82/root/usr/include/php/ext
  -DHAVE_CONFIG_H  -g -O2 -D_GNU_SOURCE    -DZEND_COMPILE_DL_EXT=1 -c /root/tmp/pear/redis/redis.c -
o redis.lo  -MMD -MF redis.dep -MT redis.lo
libtool: compile:  cc -I. -I/root/tmp/pear/redis -I/root/tmp/pear/pear-build-rootaQcBse/redis-6.1.0/
include -I/root/tmp/pear/pear-build-rootaQcBse/redis-6.1.0/main -I/root/tmp/pear/redis -I/opt/cpanel
/ea-php82/root/usr/include/php -I/opt/cpanel/ea-php82/root/usr/include/php/main -I/opt/cpanel/ea-php
82/root/usr/include/php/TSRM -I/opt/cpanel/ea-php82/root/usr/include/php/Zend -I/opt/cpanel/ea-php82
/root/usr/include/php/ext -I/opt/cpanel/ea-php82/root/usr/include/php/ext/date/lib -I/opt/cpanel/ea-
php82/root/usr/include/php/ext -DHAVE_CONFIG_H -g -O2 -D_GNU_SOURCE -DZEND_COMPILE_DL_EXT=1 -c /root
/tmp/pear/redis/redis.c -MMD -MF redis.dep -MT redis.lo  -fPIC -DPIC -o .libs/redis.o
/root/tmp/pear/redis/redis.c: In function 'zim_Redis_multi':
/root/tmp/pear/redis/redis.c:1909:58: error: macro "strncmp" requires 3 arguments, but only 2 given
                 } else if (strncmp(resp, ZEND_STRL("+OK")) != 0) {
                                                          ^
/root/tmp/pear/redis/redis.c: In function 'redis_response_enqueued':
/root/tmp/pear/redis/redis.c:2048:47: error: macro "strncmp" requires 3 arguments, but only 2 given
         if (strncmp(resp, ZEND_STRL("+QUEUED")) == 0) {
                                               ^
/root/tmp/pear/redis/redis.c: In function 'redis_sock_read_multibulk_multi_reply_loop':
/root/tmp/pear/redis/redis.c:2072:44: error: macro "strncmp" requires 3 arguments, but only 2 given
             strncmp(inbuf, ZEND_STRL("+OK")) != 0)
                                            ^
make: *** [redis.lo] Error 1
ERROR: `make' failed

It's because `strncmp` on this system is a macro.

```
/root/tmp/pear/redis/redis.c: In function 'redis_response_enqueued':
/root/tmp/pear/redis/redis.c:2048:47: error: macro "strncmp" requires 3 arguments, but only 2 given
if (strncmp(resp, ZEND_STRL("+QUEUED")) == 0) {
```

I've never seen this on a Linux system before. It's simple enough to fix although i think there are a few more `strncmp` invocations in files other than redis.c

Hello, FWIW: we're seeing these errors on RHEL7 and on RHEL8 as well.
No errors on RHEL9.

```
/tmp/pear/temp/redis/redis.c: In function 'zim_Redis_multi':
/tmp/pear/temp/redis/redis.c:1909:58: error: macro "strncmp" requires 3 arguments, but only 2 given
                 } else if (strncmp(resp, ZEND_STRL("+OK")) != 0) {
                                                          ^
/tmp/pear/temp/redis/redis.c: In function 'redis_response_enqueued':
/tmp/pear/temp/redis/redis.c:2048:47: error: macro "strncmp" requires 3 arguments, but only 2 given
         if (strncmp(resp, ZEND_STRL("+QUEUED")) == 0) {
                                               ^
/tmp/pear/temp/redis/redis.c: In function 'redis_sock_read_multibulk_multi_reply_loop':
/tmp/pear/temp/redis/redis.c:2072:44: error: macro "strncmp" requires 3 arguments, but only 2 given
             strncmp(inbuf, ZEND_STRL("+OK")) != 0)
                                            ^
make: *** [Makefile:210: redis.lo] Error 1
```

Thanks for the quick response! Almost there it seems:
```
phpredis/cluster_library.c: In function 'cluster_type_resp':
phpredis/cluster_library.c:1913:55: error: macro "strncmp" requires 3 arguments, but only 2 given
     } else if (strncmp(c->line_reply, ZEND_STRL("set")) == 0) {
                                                       ^
make: *** [Makefile:231: cluster_library.lo] Error 1
```
Just one more occurrence - after changing the following the build is working:
```
--- a/cluster_library.c
+++ b/cluster_library.c
@@ -1910,7 +1910,7 @@ PHP_REDIS_API void cluster_type_resp(INTERNAL_FUNCTION_PARAMETERS, redisCluster
     // Switch on the type
     if (redis_strncmp(c->line_reply, ZEND_STRL("string")) == 0) {
         CLUSTER_RETURN_LONG(c, REDIS_STRING);
-    } else if (strncmp(c->line_reply, ZEND_STRL("set")) == 0) {
+    } else if (redis_strncmp(c->line_reply, ZEND_STRL("set")) == 0) {
         CLUSTER_RETURN_LONG(c, REDIS_SET);
     } else if (redis_strncmp(c->line_reply, ZEND_STRL("list")) == 0) {
         CLUSTER_RETURN_LONG(c, REDIS_LIST);
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -68,7 +68,7 @@ jobs:
           grep "REDIS_SHARED_LIBADD.*-L$GITHUB_WORKSPACE/libzstd" Makefile
 
   ubuntu:
-    runs-on: ubuntu:23.04
+    runs-on: ubuntu-22.04
     continue-on-error: false
     strategy:
       fail-fast: false

--- a/cluster_library.c
+++ b/cluster_library.c
@@ -1908,17 +1908,17 @@ PHP_REDIS_API void cluster_type_resp(INTERNAL_FUNCTION_PARAMETERS, redisCluster 
     }
 
     // Switch on the type
-    if (strncmp (c->line_reply, "string", 6) == 0) {
+    if (redis_strncmp(c->line_reply, ZEND_STRL("string")) == 0) {
         CLUSTER_RETURN_LONG(c, REDIS_STRING);
-    } else if (strncmp(c->line_reply, ZEND_STRL("set")) == 0) {
+    } else if (redis_strncmp(c->line_reply, ZEND_STRL("set")) == 0) {
         CLUSTER_RETURN_LONG(c, REDIS_SET);
-    } else if (strncmp(c->line_reply, ZEND_STRL("list")) == 0) {
+    } else if (redis_strncmp(c->line_reply, ZEND_STRL("list")) == 0) {
         CLUSTER_RETURN_LONG(c, REDIS_LIST);
-    } else if (strncmp(c->line_reply, ZEND_STRL("hash")) == 0) {
+    } else if (redis_strncmp(c->line_reply, ZEND_STRL("hash")) == 0) {
         CLUSTER_RETURN_LONG(c, REDIS_HASH);
-    } else if (strncmp(c->line_reply, ZEND_STRL("zset")) == 0) {
+    } else if (redis_strncmp(c->line_reply, ZEND_STRL("zset")) == 0) {
         CLUSTER_RETURN_LONG(c, REDIS_ZSET);
-    } else if (strncmp(c->line_reply, ZEND_STRL("stream")) == 0) {
+    } else if (redis_strncmp(c->line_reply, ZEND_STRL("stream")) == 0) {
         CLUSTER_RETURN_LONG(c, REDIS_STREAM);
     } else {
         CLUSTER_RETURN_LONG(c, REDIS_NOT_FOUND);
@@ -1977,9 +1977,9 @@ PHP_REDIS_API void cluster_sub_resp(INTERNAL_FUNCTION_PARAMETERS, redisCluster *
         }
 
         // Make sure we have a message or pmessage
-        if (!strncmp(Z_STRVAL_P(z_type), ZEND_STRL("message")) ||
-            !strncmp(Z_STRVAL_P(z_type), ZEND_STRL("pmessage"))
-        ) {
+        if (zend_string_equals_literal(Z_STR_P(z_type), "message") ||
+            zend_string_equals_literal(Z_STR_P(z_type), "pmessage"))
+        {
             is_pmsg = *Z_STRVAL_P(z_type) == 'p';
         } else {
             zval_dtor(&z_tab);

--- a/common.h
+++ b/common.h
@@ -262,6 +262,12 @@ typedef enum {
 #define REDIS_STRICMP_STATIC(s, len, sstr) \
     (len == sizeof(sstr) - 1 && !strncasecmp(s, sstr, len))
 
+/* On some versions of glibc strncmp is a macro. This wrapper allows us to
+   use it in combination with ZEND_STRL in those cases. */
+static inline int redis_strncmp(const char *s1, const char *s2, size_t n) {
+    return strncmp(s1, s2, n);
+}
+
 /* Test if a zval is a string and (case insensitive) matches a static string */
 #define ZVAL_STRICMP_STATIC(zv, sstr) \
     REDIS_STRICMP_STATIC(Z_STRVAL_P(zv), Z_STRLEN_P(zv), sstr)

--- a/library.c
+++ b/library.c
@@ -147,7 +147,7 @@ static int reselect_db(RedisSock *redis_sock) {
         return -1;
     }
 
-    if (strncmp(response, ZEND_STRL("+OK"))) {
+    if (redis_strncmp(response, ZEND_STRL("+OK"))) {
         efree(response);
         return -1;
     }
@@ -247,7 +247,7 @@ PHP_REDIS_API int redis_sock_auth(RedisSock *redis_sock) {
     efree(cmd);
 
     if (redis_sock_gets(redis_sock, inbuf, sizeof(inbuf) - 1, &len) < 0 ||
-        strncmp(inbuf, ZEND_STRL("+OK")))
+        redis_strncmp(inbuf, ZEND_STRL("+OK")))
     {
         return FAILURE;
     }
@@ -1202,17 +1202,17 @@ PHP_REDIS_API int redis_type_response(INTERNAL_FUNCTION_PARAMETERS, RedisSock *r
         return FAILURE;
     }
 
-    if (strncmp(response, ZEND_STRL("+string")) == 0) {
+    if (redis_strncmp(response, ZEND_STRL("+string")) == 0) {
         l = REDIS_STRING;
-    } else if (strncmp(response, ZEND_STRL("+set")) == 0){
+    } else if (redis_strncmp(response, ZEND_STRL("+set")) == 0){
         l = REDIS_SET;
-    } else if (strncmp(response, ZEND_STRL("+list")) == 0){
+    } else if (redis_strncmp(response, ZEND_STRL("+list")) == 0){
         l = REDIS_LIST;
-    } else if (strncmp(response, ZEND_STRL("+zset")) == 0){
+    } else if (redis_strncmp(response, ZEND_STRL("+zset")) == 0){
         l = REDIS_ZSET;
-    } else if (strncmp(response, ZEND_STRL("+hash")) == 0){
+    } else if (redis_strncmp(response, ZEND_STRL("+hash")) == 0){
         l = REDIS_HASH;
-    } else if (strncmp(response, ZEND_STRL("+stream")) == 0) {
+    } else if (redis_strncmp(response, ZEND_STRL("+stream")) == 0) {
         l = REDIS_STREAM;
     } else {
         l = REDIS_NOT_FOUND;
@@ -3013,18 +3013,18 @@ redis_sock_check_liveness(RedisSock *redis_sock)
     }
 
     if (auth) {
-        if (strncmp(inbuf, ZEND_STRL("+OK")) == 0 ||
-            strncmp(inbuf, ZEND_STRL("-ERR Client sent AUTH")) == 0)
+        if (redis_strncmp(inbuf, ZEND_STRL("+OK")) == 0 ||
+            redis_strncmp(inbuf, ZEND_STRL("-ERR Client sent AUTH")) == 0)
         {
             /* successfully authenticated or authentication isn't required */
             if (redis_sock_gets(redis_sock, inbuf, sizeof(inbuf) - 1, &len) < 0) {
                 goto failure;
             }
-        } else if (strncmp(inbuf, ZEND_STRL("-NOAUTH")) == 0) {
+        } else if (redis_strncmp(inbuf, ZEND_STRL("-NOAUTH")) == 0) {
             /* connection is fine but authentication failed, next command must
              * fail too */
             if (redis_sock_gets(redis_sock, inbuf, sizeof(inbuf) - 1, &len) < 0
-                || strncmp(inbuf, ZEND_STRL("-NOAUTH")) != 0)
+                || redis_strncmp(inbuf, ZEND_STRL("-NOAUTH")) != 0)
             {
                 goto failure;
             }
@@ -3034,7 +3034,7 @@ redis_sock_check_liveness(RedisSock *redis_sock)
         }
         redis_sock->status = REDIS_SOCK_STATUS_AUTHENTICATED;
     } else {
-        if (strncmp(inbuf, ZEND_STRL("-NOAUTH")) == 0) {
+        if (redis_strncmp(inbuf, ZEND_STRL("-NOAUTH")) == 0) {
             /* connection is fine but authentication required */
             return SUCCESS;
         }
@@ -3042,11 +3042,11 @@ redis_sock_check_liveness(RedisSock *redis_sock)
 
     /* check echo response */
     if ((redis_sock->sentinel && (
-        strncmp(inbuf, ZEND_STRL("-ERR unknown command")) != 0 ||
+        redis_strncmp(inbuf, ZEND_STRL("-ERR unknown command")) != 0 ||
         strstr(inbuf, id) == NULL
     )) || *inbuf != TYPE_BULK || atoi(inbuf + 1) != idlen ||
         redis_sock_gets(redis_sock, inbuf, sizeof(inbuf) - 1, &len) < 0 ||
-        strncmp(inbuf, id, idlen) != 0
+        redis_strncmp(inbuf, id, idlen) != 0
     ) {
         goto failure;
     }

--- a/redis.c
+++ b/redis.c
@@ -174,7 +174,7 @@ redis_send_discard(RedisSock *redis_sock)
        (resp = redis_sock_read(redis_sock,&resp_len)) != NULL)
     {
         /* success if we get OK */
-        result = (resp_len == 3 && strncmp(resp,"+OK", 3) == 0) ? SUCCESS:FAILURE;
+        result = (resp_len == 3 && redis_strncmp(resp, ZEND_STRL("+OK")) == 0) ? SUCCESS:FAILURE;
 
         /* free our response */
         efree(resp);
@@ -1906,7 +1906,7 @@ PHP_METHOD(Redis, multi)
                 }
                 if ((resp = redis_sock_read(redis_sock, &resp_len)) == NULL) {
                     RETURN_FALSE;
-                } else if (strncmp(resp, ZEND_STRL("+OK")) != 0) {
+                } else if (redis_strncmp(resp, ZEND_STRL("+OK")) != 0) {
                     efree(resp);
                     RETURN_FALSE;
                 }
@@ -2045,7 +2045,7 @@ redis_response_enqueued(RedisSock *redis_sock)
     int resp_len, ret = FAILURE;
 
     if ((resp = redis_sock_read(redis_sock, &resp_len)) != NULL) {
-        if (strncmp(resp, ZEND_STRL("+QUEUED")) == 0) {
+        if (redis_strncmp(resp, ZEND_STRL("+QUEUED")) == 0) {
             ret = SUCCESS;
         }
         efree(resp);
@@ -2069,7 +2069,7 @@ redis_sock_read_multibulk_multi_reply_loop(INTERNAL_FUNCTION_PARAMETERS,
         char inbuf[255];
 
         if (redis_sock_gets(redis_sock, inbuf, sizeof(inbuf) - 1, &len) < 0 ||
-            strncmp(inbuf, ZEND_STRL("+OK")) != 0)
+            redis_strncmp(inbuf, ZEND_STRL("+OK")) != 0)
         {
             return FAILURE;
         }

--- a/redis_array_impl.c
+++ b/redis_array_impl.c
@@ -139,7 +139,7 @@ ra_find_name(const char *name) {
     for(p = ini_names; p;) {
         next = strchr(p, ',');
         if(next) {
-            if(strncmp(p, name, next - p) == 0) {
+            if(redis_strncmp(p, name, next - p) == 0) {
                 return 1;
             }
         } else {
@@ -283,7 +283,7 @@ RedisArray *ra_load_array(const char *name) {
         sapi_module.treat_data(PARSE_STRING, estrdup(iptr), &z_tmp);
         if ((z_data = zend_hash_str_find(Z_ARRVAL(z_tmp), name, name_len)) != NULL) {
             consistent = Z_TYPE_P(z_data) == IS_STRING &&
-                         strncmp(Z_STRVAL_P(z_data), ZEND_STRL("1")) == 0;
+                         redis_strncmp(Z_STRVAL_P(z_data), ZEND_STRL("1")) == 0;
         }
         zval_dtor(&z_tmp);
     }

--- a/redis_session.c
+++ b/redis_session.c
@@ -336,7 +336,7 @@ static int lock_acquire(RedisSock *redis_sock, redis_session_lock_status *lock_s
     return lock_status->is_locked ? SUCCESS : FAILURE;
 }
 
-#define IS_LOCK_SECRET(reply, len, secret) (len == ZSTR_LEN(secret) && !strncmp(reply, ZSTR_VAL(secret), len))
+#define IS_LOCK_SECRET(reply, len, secret) (len == ZSTR_LEN(secret) && !redis_strncmp(reply, ZSTR_VAL(secret), len))
 static int write_allowed(RedisSock *redis_sock, redis_session_lock_status *lock_status)
 {
     if (!INI_INT("redis.session.locking_enabled")) {
@@ -444,7 +444,7 @@ PS_OPEN_FUNC(redis)
             zend_string *user = NULL, *pass = NULL;
 
             /* translate unix: into file: */
-            if (!strncmp(save_path+i, "unix:", sizeof("unix:")-1)) {
+            if (!redis_strncmp(save_path+i, ZEND_STRL("unix:"))) {
                 int len = j-i;
                 char *path = estrndup(save_path+i, len);
                 memcpy(path, "file:", sizeof("file:")-1);

